openedx_external_enrollments/tests/api/v0/test_views.py

```diff
@@ -0,0 +1,689 @@
+        self.assertEqual(self.base._get_enrollment_headers(), expected_headers)
+
+    def test_get_enrollment_url(self):
+        """Testing _get_enrollment_url method."""
+        expected_url = 'https://edx-external-instance.com/api/v0/enrollments'
+        course_settings = {
+            'external_enrollment_api_url': expected_url
+        }
+        self.assertIsNone(
+            self.base._get_enrollment_url(course_settings={}),
+        )
+
+        self.assertEqual(
+            expected_url,
+            self.base._get_enrollment_url(course_settings),
+        )
+
+    def test_str(self):
+        """
+        EdxInstanceExternalEnrollment overrides the __str__ method,
+        this test that the method __str__ returns the right value.
+        """
+        self.assertEqual(
+            'openedX',
+            self.base.__str__(),
+        )
+
+
+class SalesforceEnrollmentTest(TestCase):
+    """Test class for SalesforceEnrollment class."""
+
+    def setUp(self):
+        """Set test database."""
+        self.base = SalesforceEnrollment()
+
+    @patch('openedx_external_enrollments.api.v0.views.get_course_by_id')
+    @patch('openedx_external_enrollments.api.v0.views.CourseKey')
+    def test_get_course(self, course_key_mock, get_course_by_id_mock):
+        """Testing _get_course method."""
+        self.assertIsNone(
+            self.base._get_course(''),
+        )
+
+        course_id = 'test-course-id'
+        course_key = 'test-course-key'
+        expected_course = 'test-course'
+        course_key_mock.from_string.return_value = course_key
+        get_course_by_id_mock.return_value = expected_course
+
+        self.assertEqual(self.base._get_course(course_id), expected_course)
+        course_key_mock.from_string.assert_called_once_with(course_id)
+        get_course_by_id_mock.assert_called_once_with(course_key)
+
+    def test_str(self):
+        """
+        SalesforceEnrollment overrides the __str__ method,
+        this test that the method __str__ returns the right value.
+        """
+        self.assertEqual(
+            'salesforce',
+            self.base.__str__(),
+        )
+
+    @patch.object(SalesforceEnrollment, '_get_auth_token')
+    def test_get_enrollment_headers(self, get_auth_token_mock):
+        """Testing _get_enrollment_headers method."""
+
+        get_auth_token_mock.return_value = {
+            'token_type': 'test-token-type',
+            'access_token': 'test-access-token',
+        }
+        expected_headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'test-token-type test-access-token',
+        }
+        self.assertEqual(self.base._get_enrollment_headers(), expected_headers)
+        get_auth_token_mock.assert_called_once()
+
+    @patch('openedx_external_enrollments.api.v0.views.OAuth2Session')
+    @patch('openedx_external_enrollments.api.v0.views.BackendApplicationClient')
+    def test_get_auth_token(self, backend_mock, oauth_session_mock):
```

Comment:
we could  have some side effects but it doesn't make sense when the method doesn't handle the exception

Reply:
Should we rewrite the method in a subsequent PR? I mean the purpose of the unit testing is to check that the code is behaving as we expected and are we expecting an exception here? @andrey-canon

Reply:
yes, you are right about the purpose, the issue is that I don't know what is expected here, maybe the intention is to raise the exception instead of handle